examples/glue/run_pl_glue.py

```diff
@@ -0,0 +1,219 @@
+                    pad_token=self.tokenizer.convert_tokens_to_ids([self.tokenizer.pad_token])[0],
+                    pad_token_segment_id=4 if args.model_type in ["xlnet"] else 0,
+                )
+                logger.info("Saving features into cached file %s", cached_features_file)
+                torch.save(features, cached_features_file)
+
+    def load_dataset(self, mode, batch_size):
+        "Load datasets. Called after prepare data."
+
+        # We test on dev set to compare to benchmarks without having to submit to GLUE server
+        mode = "dev" if mode == "test" else mode
+
+        cached_features_file = self._feature_file(mode)
+        logger.info("Loading features from cached file %s", cached_features_file)
+        features = torch.load(cached_features_file)
+        all_input_ids = torch.tensor([f.input_ids for f in features], dtype=torch.long)
+        all_attention_mask = torch.tensor([f.attention_mask for f in features], dtype=torch.long)
+        all_token_type_ids = torch.tensor([f.token_type_ids for f in features], dtype=torch.long)
+        if self.hparams.glue_output_mode == "classification":
+            all_labels = torch.tensor([f.label for f in features], dtype=torch.long)
+        elif self.hparams.glue_output_mode == "regression":
+            all_labels = torch.tensor([f.label for f in features], dtype=torch.float)
+
+        return DataLoader(
+            TensorDataset(all_input_ids, all_attention_mask, all_token_type_ids, all_labels),
+            batch_size=batch_size,
+            shuffle=True,
+        )
+
+    def validation_step(self, batch, batch_idx):
+        inputs = {"input_ids": batch[0], "attention_mask": batch[1], "labels": batch[3]}
+
+        if self.hparams.model_type != "distilbert":
+            inputs["token_type_ids"] = batch[2] if self.hparams.model_type in ["bert", "xlnet", "albert"] else None
+        outputs = self.forward(**inputs)
+        tmp_eval_loss, logits = outputs[:2]
+        preds = logits.detach().cpu().numpy()
+        out_label_ids = inputs["labels"].detach().cpu().numpy()
+
+        return {"val_loss": tmp_eval_loss.detach().cpu(), "pred": preds, "target": out_label_ids}
+
+    def _eval_end(self, outputs):
+        val_loss_mean = torch.stack([x["val_loss"] for x in outputs]).mean().detach().cpu().item()
+        preds = np.concatenate([x["pred"] for x in outputs], axis=0)
+
+        if self.hparams.glue_output_mode == "classification":
+            preds = np.argmax(preds, axis=1)
+        elif self.hparams.glue_output_mode == "regression":
+            preds = np.squeeze(preds)
+
+        out_label_ids = np.concatenate([x["target"] for x in outputs], axis=0)
+        out_label_list = [[] for _ in range(out_label_ids.shape[0])]
+        preds_list = [[] for _ in range(out_label_ids.shape[0])]
+
+        results = {**{"val_loss": val_loss_mean}, **compute_metrics(self.hparams.task, preds, out_label_ids)}
+
+        if self.is_logger():
```

Comment:
might want to wrap up all this other code in a callback.
https://pytorch-lightning.readthedocs.io/en/0.7.1/callbacks.html

this way, you don't polute the research code with all of this non-essential code.

Reply:
Yup, This makes sense to me. I added this before you could have multiple loggers. @nateraw if you can figure out how, I think we should default to a TensorBoard logged and a CLI logger for each round.

Reply:
@srush How does [this](https://github.com/huggingface/transformers/pull/3290/commits/82546a7e5a37efc20f264a843008ac92ebd21298) look?

Reply:
I don't like this either. How about just add this to the .sh

`export PYTHONPATH="../":"${PYTHONPATH}"`

Reply:
This looks good to me. The only issue is that I think this clobbers the default TensorBoard logger. Let's pass that in by default as well.

Reply:
@williamFalcon is this true? It was my understanding that the `callbacks` kwarg was separate

Reply:
the loggers and explicit callbacks don't get overriden if we pass in other callbacks. Sorry about the confusion, we'll clear up the docs.

Reply:
ie:
EarlyStopping + Loggers + Checkpoint + other callbacks

NOT
callbacks OR (EarlyStopping + Loggers + Checkpoint)

Reply:
i tried to look at "this" in the link but it's not clear. I assume this is what you're asking about?

Reply:
Yes, you cleared things up. Thanks!